packages/alfa-dom/src/matches.ts

```diff
@@ -430,12 +430,24 @@ function matchesPseudoClass(
 
     // https://drafts.csswg.org/css-scoping/#host-selector
     case "host":
+      // Do not allow prefix (e.g. "div:host")
+      if (root !== selector) {
+        return false;
+      }
+
+      if (options.treeContext === undefined) {
```

Comment:
This should also check for `isShadowRoot(options.treeContext)` so as to bail out early if the tree context isn't a shadow root.